go/libraries/doltcore/sqle/tables.go

```diff
@@ -2408,7 +2408,43 @@ func parseFkReferentialAction(refOp sql.ForeignKeyReferentialAction) (doltdb.For
 // dropIndex drops the given index on the given table with the given schema. Returns the updated table and updated schema.
 func (t *AlterableDoltTable) dropIndex(ctx *sql.Context, indexName string) (*doltdb.Table, schema.Schema, error) {
 	// RemoveIndex returns an error if the index does not exist, no need to do twice
-	_, err := t.sch.Indexes().RemoveIndex(indexName)
+	oldIdx, err := t.sch.Indexes().RemoveIndex(indexName)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// any foreign keys that used this underlying index need to find another one
+	root, err := t.getRoot(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+	fkc, err := root.GetForeignKeyCollection(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+	for _, fk := range fkc.AllKeys() {
+		newIdx, ok, err := findIndexWithPrefix(t.sch, oldIdx.ColumnNames())
+		if err != nil {
+			return nil, nil, err
+		}
+		newFk := fk
+		if ok {
+			newFk.ReferencedTableIndex = newIdx.Name()
+		} else {
+			// if a replacement index wasn't found; it matched on primary key, so use empty string
+			newFk.ReferencedTableIndex = ""
```

Comment:
I think we can make this assumption, because we do in other places like [here](https://github.com/dolthub/dolt/blob/main/go/libraries/doltcore/sqle/tables.go#L2207) and [here](https://github.com/dolthub/dolt/blob/main/go/libraries/doltcore/sqle/tables.go#L2241).

This is because GMS also checks if a valid prefix index exists, and errors when there isn't (I updated the tests to confirm this behavior). In the case of switching to a PK as the underlying index, GMS actually considers PKs as indexes and will correctly not error. However, on dolt side, we don't treat PK as an index, so `findIndexWithPrefix` wont be able to find it. A side effect of this is that if there is a viable PK and multiple viable secondary indexes, the PK will only be chosen if we drop all other viable secondary indexes first. 

GMS Code for reference:
 - [drop index] (https://github.com/dolthub/go-mysql-server/blob/main/sql/plan/alter_index.go#L251)
 - [find prefix] (https://github.com/dolthub/go-mysql-server/blob/main/sql/plan/alter_foreign_key.go#L525)